### PR TITLE
GH-115983: skip building shared modules for testing under WASI

### PR DESCRIPTION
**Pull request by brettcannon**
* Issue: gh-115983

### CODE REVIEW

**Comment by brettcannon** [comment]
!buildbot wasi

**Comment by erlend-aasland** [comment]
Could you instead add these using the `PY_STDLIB_MOD_SET_NA` macro?

https://github.com/python/cpython/blob/b9cb855621c0813846421e4ced97260a32e57063/configure.ac#L7333-L7410

**Comment by brettcannon** [comment]
> Could you instead add these using the `PY_STDLIB_MOD_SET_NA` macro?

Done!

File: configure.ac [hunk 1]
```diff
@@ -7398,11 +7398,19 @@ AS_CASE([$ac_sys_system],
...
       [WASI/*], [
         dnl WASI SDK 15.0 does not support file locking, mmap, and more.
+        dnl Test modules that must be compiled as shared libraries are not supported
+        dnl (see Modules/Setup.stdlib.in).
         PY_STDLIB_MOD_SET_NA(
           [_ctypes_test],
+          [_testexternalinspection],
+          [_testimportmultiple],
+          [_testmultiphase],
+          [_testsinglephase],
```
Comment (erlend-aasland) [inline configure.ac:7408]: What about the rest of the `_test*` modules? There are nine :)

Comment (brettcannon) [reply configure.ac:7408]: They can be compiled in statically and thus aren't a concern. The list in this PR is specific to:

https://github.com/python/cpython/blob/149f7f7ae28944579792d22607532006977177c9/Modules/Setup.stdlib.in#L170-L180

which must be explicitly compiled as shared modules.

Comment (erlend-aasland) [reply configure.ac:7408]: Ok, thanks!

**Review by erlend-aasland — APPROVED** [review]

**Comment by brettcannon** [comment]
FYI I purposefully didn't bothering backporting as tier 2 support was only for 3.13 and I have made no guarantees older versions would work with newer WASI SDK versions (there's actually a discussion to be had if they even should as binary wheels wouldn't be compatible w/ different SDK versions).

**Comment by vstinner** [comment]
> FYI I purposefully didn't bothering backporting as tier 2 support was only for 3.13

The backport was needed to repair the WASI buildbots: regression caused by https://github.com/python/cpython/pull/120173 backport to 3.12.

**Pull request merged**